pkg/instancetype/controller/vm/controller_test.go

```diff
@@ -0,0 +1,1052 @@
+		vm = libvmi.NewVirtualMachine(vmi)
+
+		// We need to clear the domainSpec here to ensure the instancetype doesn't conflict
+		vm.Spec.Template.Spec.Domain = virtv1.DomainSpec{}
+
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		virtClient.EXPECT().VirtualMachine(metav1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().KubevirtV1().VirtualMachines(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineInstance(metav1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineInstancetype(k8sv1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachineInstancetypes(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineClusterInstancetype().Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachineClusterInstancetypes()).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachinePreference(k8sv1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachinePreferences(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineClusterPreference().Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachineClusterPreferences()).AnyTimes()
+
+		virtClient.EXPECT().AppsV1().Return(k8sfake.NewSimpleClientset().AppsV1()).AnyTimes()
+
+		instancetypeInformer, _ := testutils.NewFakeInformerFor(&v1beta1.VirtualMachineInstancetype{})
+		instancetypeInformerStore = instancetypeInformer.GetStore()
+
+		clusterInstancetypeInformer, _ := testutils.NewFakeInformerFor(&v1beta1.VirtualMachineClusterInstancetype{})
+		clusterInstancetypeInformerStore = clusterInstancetypeInformer.GetStore()
+
+		preferenceInformer, _ := testutils.NewFakeInformerFor(&v1beta1.VirtualMachinePreference{})
+		preferenceInformerStore = preferenceInformer.GetStore()
+
+		clusterPreferenceInformer, _ := testutils.NewFakeInformerFor(&v1beta1.VirtualMachineClusterPreference{})
+		clusterPreferenceInformerStore = clusterPreferenceInformer.GetStore()
+
+		controllerrevisionInformer, _ := testutils.NewFakeInformerFor(&appsv1.ControllerRevision{})
+		controllerrevisionInformerStore = controllerrevisionInformer.GetStore()
+
+		config, _, kvStore = testutils.NewFakeClusterConfigUsingKVConfig(&virtv1.KubeVirtConfiguration{})
+
+		recorder = record.NewFakeRecorder(100)
+		recorder.IncludeObject = true
+
+		instancetypeController = instancetypecontroller.New(
+			instancetypeInformerStore,
+			clusterInstancetypeInformerStore,
+			preferenceInformerStore,
+			clusterPreferenceInformerStore,
+			controllerrevisionInformerStore,
+			virtClient,
+			config,
+			recorder,
+		)
+
+		instancetypeObj = &v1beta1.VirtualMachineInstancetype{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: v1beta1.SchemeGroupVersion.String(),
+				Kind:       "VirtualMachineInstancetype",
+			},
```

Comment:
nit: Probably can drop this

Reply:
We can't as it's used by `decodeControllerRevisionObject` and `runtime.Decode` in the compat code:

https://github.com/lyarwood/kubevirt/blob/8f27d9020e7a891dda88068339b9e229e9406bc2/pkg/instancetype/compatibility/compatibility.go